.gitignore

```diff
@@ -6,4 +6,19 @@ __pycache__/
 *.swp
 *~
 
+# cookie cutter
+{{cookiecutter.project_name}}/requirements-dev.txt
```

Comment:
I'm not a fan of this multiple requirements pattern. It's a slightly faster build time in exchange for a slighy difference in environments and an extra file in the repo. I appreciate it's now the defacto in the front end world. Happy to be outvoted on this one. Interested in what everyone else thinks.

Reply:
I don't have a strong argument for or against, imo it's tidier and as you say removes some bloat on deployment/potentially removes some chance of developer error like leaving debug tools enabled by mistake etc but those aren't huge reasons, so happy to go with the consensus. 

I actually really like the way DRF does it https://github.com/encode/django-rest-framework/tree/master/requirements but guessing that is even worse from your perspective!

Reply:
I've been playing with this myself, I have requirements.txt with

```
-r requirements-server.txt
-r requirements-test.txt
```

and both of those get built by pip-tools. In my main container I install `requirements-server.txt` and I have a test container that extends the server that also installs `requirements-test.txt`, for dev your workflow doesn't change and you just install `requirements.txt` / `pip-sync`.

The only issue with this approach is if there is a dependency mismatch, for instance if package `A` requires and version of `B` in the server reqs but the `C` requires `B<X` there will be a miss match when trying to install `requirements.txt` but this then just requires adding the restriction to both `.in` files.

Personally I think it's worth it for both image size and build time.

Reply:
I lean gently towards splitting dev out. On PHE NSCR we removed quite a bit of bloat - 52 deployed packages against 125 dev packages, which seems sensible.

Almost more importantly, it forces a decision about where a package goes, and so should help clarify how important things are when it comes to managing upgrades etc.

Main negative it's also easy to add a package to the wrong requirement in dev, and wonder why it's failing on CI.

I like DRF's idea of grouping requirements, but think it'd be more manageable to keep them in the one file, then group them and explain their purpose with comments.

Agree with Rolo about ending up with more files though - when you end up with lots of requirements it can get confusing. Think it's worth keeping the number of requirements files to a minimum, eg `requirements.in` and `requirements-dev.in` (which itself includes `requirements.in`, to enforce dev consistency with deployed).

Reply:
Id be fine with that other than changing it so dev is in `requirements.in` and there is a `requirements-deployed.in` that way when devving there's nothing special and installing specific reqs file is stored in script.

Reply:
As discussed, single in for now added.